src/inspector/node_protocol.pdl

```diff
@@ -0,0 +1,50 @@
+# Please notify @nodejs/v8-inspector and @nodejs/trace-events before modifying this file
+version
+  major 1
+  minor 0
+
+experimental domain NodeTracing
+  type TraceConfig extends object
+    properties
+      # Controls how the trace buffer stores data.
+      optional enum recordMode
+        recordUntilFull
+        recordContinuously
+        recordAsMuchAsPossible
+      # Included category filters.
+      array of string includedCategories
+
+  # Stop trace events collection.
+  command end
+
+  # Gets supported tracing categories.
+  command getCategories
+    returns
+      # A list of supported tracing categories.
+      array of string categories
+
+  # Start trace events collection.
+  command start
+    parameters
+      TraceConfig traceConfig
+
+  event bufferUsage
+    parameters
+      # A number in range [0..1] that indicates the used size of event buffer as a fraction of its
+      # total size.
+      optional number percentFull
+      # An approximate number of events in the trace log.
+      optional number eventCount
+      # A number in range [0..1] that indicates the used size of event buffer as a fraction of its
+      # total size.
+      optional number value
+
+  # Contains an bucket of collected trace events. When tracing is stopped collected events will be
```

Comment:
I just want to clarify: with this channel, are the events buffered in Node.js until a given point and then delivered or are they streamed continuously as they are emitted? The latter would be far more useful in the long run. If it's the former then it will be good to know :-)

Reply:
@a1ph Did that.

@jasnell Right now events are sent out when trace buffer is flushed, same as writing the trace to the file. This is done to minimize performance impact (and also not to brake the trace to file). Some events will force flush, such as connecting/disconnecting an inspector session (multiple sessions are supported) or using JS API to tweak enabled categories.

Reply:
Just thinking out load... I wonder if it would make sense to support some signal to force flushing on demand.